### PR TITLE
IDEX-3295 Select Git importer by default

### PR DESCRIPTION
**Pull request by mkuznyetsov**
Do not merge until properties are added

### CODE REVIEW

File: ide/che-core-ide-app/src/main/java/org/eclipse/che/ide/projectimport/wizard/mainpage/MainPagePresenter.java [hunk 1]
```diff
@@ -140,13 +141,20 @@ protected void onSuccess(List<ProjectImporterDescriptor> result) {
                                 importersSet.add(importer);
                                 importersByCategory.put(importer.getCategory(), importersSet);
                             }
+
+                            if (importer.getId().equals("git")) {
```
Comment (vzhukovskii) [inline ide/che-core-ide-app/src/main/java/org/eclipse/che/ide/projectimport/wizard/mainpage/MainPagePresenter.java]: I think this should be configurable. @vparfonov wdyt?

**Comment by vparfonov** [comment]
totally agree with @vzhukovskii hard-code here not good solution. Maybe better solution make in with some properties that come form server side. For example for now we return form server just  listProjectImporerDescription but we can add options with some properties like "defautlImporerId"

**Comment by ashumilova** [comment]
+1

File: ide/che-core-ide-app/src/main/java/org/eclipse/che/ide/projectimport/wizard/mainpage/MainPagePresenter.java [hunk 2]
```diff
@@ -118,15 +122,29 @@ public void go(final AcceptsOneWidget container) {
...
-        AsyncRequestCallback<List<ProjectImporterDescriptor>> callback =
-                new AsyncRequestCallback<List<ProjectImporterDescriptor>>(unmarshaller) {
+        AsyncRequestCallback<ProjectImporterData> callback =
+                new AsyncRequestCallback<ProjectImporterData>(unmarshaller) {
                     @Override
-                    protected void onSuccess(List<ProjectImporterDescriptor> result) {
+                    protected void onSuccess(ProjectImporterData data) {
+                        List<ProjectImporterDescriptor> result = data.getImporters();
+                        String defaultImporterId = data.getConfiguration().get(DEFAULT_PROJECT_IMPORTER);
+                        Iterator<ProjectImporterDescriptor> itr = result.iterator();
```
Comment (vparfonov) [inline ide/che-core-ide-app/src/main/java/org/eclipse/che/ide/projectimport/wizard/mainpage/MainPagePresenter.java]: why two loops here one with iterator other  with "for (ProjectImporterDescriptor importer : result)" ?

File: platform-api/che-core-api-project/src/main/java/org/eclipse/che/api/project/server/ProjectImportersService.java [hunk 3]
```diff
@@ -31,19 +37,23 @@
 
     private final ProjectImporterRegistry importersRegistry;
 
+    private final Map<String, String> configuration;
+
     @Inject
-    public ProjectImportersService(ProjectImporterRegistry importersRegistry) {
+    public ProjectImportersService(ProjectImporterRegistry importersRegistry,
+                                   @Named("project.importer.default_importer_id") String defaultProjectImporter) {
```
Comment (vparfonov) [inline platform-api/che-core-api-project/src/main/java/org/eclipse/che/api/project/server/ProjectImportersService.java]: maybe project.importer.default_importer_id  must be optional?

File: ide/che-core-ide-app/src/main/java/org/eclipse/che/ide/projectimport/wizard/mainpage/MainPagePresenter.java [hunk 4]
```diff
@@ -136,24 +154,30 @@ protected void onSuccess(List<ProjectImporterDescriptor> result) {
...
                                 importersByCategory.get(importer.getCategory()).add(importer);
                             } else {
-                                Set<ProjectImporterDescriptor> importersSet = new HashSet<>();
+                                Set<ProjectImporterDescriptor> importersSet = new LinkedHashSet<>();
                                 importersSet.add(importer);
                                 importersByCategory.put(importer.getCategory(), importersSet);
                             }
+
+                            if (importer.getId().equals(defaultImporterId)) {
+                                    defaultImporter = importer;
```
Comment (azatsarynnyy) [inline ide/che-core-ide-app/src/main/java/org/eclipse/che/ide/projectimport/wizard/mainpage/MainPagePresenter.java]: not foramatted line - one extra tab

File: ide/che-core-ide-app/src/main/java/org/eclipse/che/ide/projectimport/wizard/mainpage/MainPagePresenter.java (hunk 2 above)
Comment (mkuznyetsov) [inline ide/che-core-ide-app/src/main/java/org/eclipse/che/ide/projectimport/wizard/mainpage/MainPagePresenter.java]: first we need to find default importer and put it and it's category into result map at first position, then we'll handle the rest of importers.

**Comment by vparfonov** [comment]
ok

**Pull request merged**